### PR TITLE
Adds method to retrieve User by an accessToken

### PR DESCRIPTION
**Pull request by bjoernHeneka**
I added the method for getting an User by accessToken. 
Initially i had the idea to rename the method getUser -> adminGetUser but there will be some downward compatibility problems for sure. Or what do you think @pmill

### CODE REVIEW

**Comment by bjoernHeneka** [comment]
Hey @pmill would it be possible to merge this?

File: src/CognitoClient.php [hunk 1]
```diff
@@ -201,6 +202,25 @@ public function confirmUserRegistration($confirmationCode, $username)
...
     }
 
+    /**
+     * @param string $accessToken
+     * @throws Exception
+     * @throws TokenExpiryException
+     * @throws TokenVerificationException
+     * @return AwsResult
+     */
+    public function getUserByToken($accessToeken)
```
Comment (pmill) [inline src/CognitoClient.php]: Looks like you've got a spelling mistake here.

File: examples/getUserByToken.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+$client->deleteUser($accessToken);
+
+
+try {
+    $user = $client->getUserByToken($username);
```
Comment (pmill) [inline examples/getUserByToken.php]: `$username` isn't declared anywhere, should it be `$accessToken` instead?

File: examples/getUserByToken.php [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+$client->deleteUser($accessToken);
```
Comment (pmill) [inline examples/getUserByToken.php]: Does this example need this line?

File: examples/getUserByToken.php [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+$accessToken = "eygresdfiOiIyVEZ1QVZ0N3JBZEJHSEF6UTB3Z3BvSVJEdXc1SnEwUkt6emJBczNpZ28wPSIsImFsZyI6IlJTMjU2In0.eyJzdWIiOiIwZGU5ZjA1NS1mMmZmLTQwNDAtODk0ZS03NDk54nk2k3jlskNDIiLCJjb2duaXRvOmdyb3VwcyI6WyJST0xFX1NVUEVSQURNSU5fVVNFUiIsIlJPTEVfQkFER0VfVVNFUiIsIlJPTEVfQURNSU5fVVNFUiIsIlJPTEVfQkFER0VfQURNSU5fVVNFUiIsIlJPTEVfU0FWRVNUT19BRE1JTl9VU0VSIiwiUk9MRV9TQVZFU1RPX1VTRVIiXSwiZXZlbnRfaWQiOiIxMTkyYzEwYS1mYjE1LTQ3MDctYWFhYy05NWUxNjdhMzY2MzUiLCJ0b2tlbl91c2UiOiJhY2Nlc3MiLCJzY29wZSI6ImF3cy5jb2duaXRvLnNpZ25pbi51c2VyLmFkbWluIiwiYXV0aF90aW1lIjoxNTc3MTg0MzMzLCJpc3MiOiJodHRwczpcL1wvY29nbml0by1pZHAuZXUtY2VudHJhbC0xLmFtYXpvbmF3cy5jb21cL2V1LWNlbnRyYWwtMV9PMnU5UFZtTzQiLCJleHAiOjE1NzcxODc5MzMsImlhdCI6MTU3NzE4NDMzNCwianRpIjoiYzc3ZDE1NGYtOWE4NS00NmI3LTkwYzYtYTRkZWY3ZmRmMjhlIiwiY2xpZW50X2lkIjoiMjg4cGQ3ZGRscGd2dGEwNG1xYzh2ZzQ0ZmYiLCJ1c2VybmFtZSI6InJlcG9ydF9hZG1pbiJ9.cCuTvOmpJ9dbSUlPfZvm9YBDMGlADxApX09ary_V7vLtTf_bn4k2vPW8ob8cD8X2ueOO9oQNEEIr1azp9yvvywT5CHl4a4Y8UeGQ_T0ZIwejRl4wf8pme-bDeU_vnWj6M2ywg_BEa7bU43qsZGwtq3lj0Pcti5VqoPJzE3qKWSqlSWsX2E3EKnPdeAheX9p-YPHDpitqSMdXluSLjrrshoevZca4NJqz8eQVHFNplVZV_6ELRhRrMmesIF_nBXC9xWwC5kU7BweWF3AVI4OILl0ZdVmV3ZWL27F77IX3Pk6iVGC7VJlgrfCa9vkGZtfKIGd1hs8cpovRN4cQgdPIRg";
```
Comment (pmill) [inline examples/getUserByToken.php]: Is this one of your actual access tokens? Might be best to this variable defined as `$accessToken = "";`, maybe with a comment above explaining how to obtain an access token.

**Review by pmill — CHANGES_REQUESTED** [review]
Hi, I just popped a couple of comments on some of the changes.

File: examples/getUserByToken.php (hunk 4 above)
Comment (bjoernHeneka) [reply examples/getUserByToken.php]: It's not a valid one.... But good idea

**Comment by bjoernHeneka** [comment]
Hey @pmill Thanks for feedback... sometimes it would be better to read the own code first 🤣

**Review by pmill — APPROVED** [review]

**Pull request merged**